gwcs/wcs.py

```diff
@@ -547,3 +551,98 @@ def footprint(self, bounding_box=None, center=False):
             vertices = _toindex(vertices)
         result = self.__call__(*vertices, **{'with_bounding_box': False})
         return np.asarray(result)
+
+    # APE 14 implementation
+
+    @property
+    def pixel_n_dim(self):
+        return self.input_frame.naxes
+
+    @property
+    def world_n_dim(self):
+        return self.output_frame.naxes
+
+    @property
+    def pixel_shape(self):
+        return None
+
+    @property
+    def pixel_bounds(self):
+        return self.bounding_box
+
+    @property
+    def world_axis_physical_types(self):
+        # Need to match with UCD type
+        return self.output_frame.axes_type
+
+    @property
+    def world_axis_units(self):
+        return [unit.to_string() for unit in self.output_frame.unit]
+
+    @property
+    def axis_correlation_matrix(self):
+        from astropy.modeling import separable
+        return separable.is_separable(self.forward_transform)
```

Comment:
I dont think this is correct, the output should be a matrix. The closes I could cook up is:
```python
s = separable.is_separable(spatial & Identity(1))
sep = np.identity(len(s), dtype=bool)
for i, a in enumerate(s):
    sep[i,i] = not(s[i])
```

Reply:
Right. I will have to work on this. It isn't ready for review yet, especially thie property.
that said comment would be very useful.

Reply:
sure, I actually came here looking for how you worked this out for compoundmodels in general and ended up getting a little distracted by matrices lol